### ISSUE TITLE
Can't connect from 24 to 25 node when erl_epmd_port is used

### ISSUE THREAD

**Issue by gomoripeti** [erlang/otp#6595]
**Describe the bug**
When I start an OTP 24 node with `-erl_epmd_port` flag defined, it failes to connect to an OTP 25 node. This is because `erl_epdm:port_please/3` does not request the dist version of the remote node from EPMD, but returns the default lowest version, which is 5 for OTP 24. However OTP 25 stopped supporting dist version 5 and only supports version 6, so the remote node does not recognise the old send_name message.

**To Reproduce**
% use OTP 25
```
erl -sname otp25@localhost
```

% in another shell use OTP 24
```
erl -sname otp24@localhost -dist_listen false -erl_epmd_port `epmd -names | grep otp25 | sed 's/name otp25 at port //'`
(otp24@localhost)1> net_kernel:connect_node('otp25@localhost').
false
```

**Expected behavior**
I expected that nodes within 2 major version difference can connect to each other independent from which node initiates the connection and no matter if EPMD is used or not.

**Affected versions**
Only tested with OTP 24.2.1 to OTP 25.0, but looking at 24.3.4.6 (compared to 24.2.1) and current maint (compared to 25.0) the code looks the same.

**Additional context**
Wonder if this is a bug or I missed the documentation of a limitation. ([`erl_epmd:port_please/3`](https://www.erlang.org/doc/man/erl_epmd.html#port_please-3) does not mention that OTP 25 returns version 6)

**Comment by sverker** [comment]
> Wonder if this is a bug or I missed the documentation of a limitation. ([erl_epmd:port_please/3](https://www.erlang.org/doc/man/erl_epmd.html#port_please-3) does not mention that OTP 25 returns version 6)

It could be considered a documentation bug, as it's possible work around by letting [erl_epmd:port_please/3](https://www.erlang.org/doc/man/erl_epmd.html#port_please-3) return 6 for OTP 25 nodes.

It can also be considered a bug in the handshake logic in lib/kernel/src/dist_util.erl, as it is fixable to accept the old send_name message which I have done in #6625.

**Comment by sverker** [comment]
@gomoripeti Thanks for a very good report by the way.

**Comment by sverker** [comment]
#6625 merged to maint for OTP 25.3.
